plugins/UsersManager/Controller.php

```diff
@@ -150,6 +150,13 @@ public function index()
         }
 
         $view->activatedPlugins = $this->pluginManager->getActivatedPlugins();
+
+        $componentExtensions = [];
```

Comment:
this should at least be prefilled with the array keys, so it's clear to the event consumer what can be overwritten. And this should also satisfy PHPStan ;-)
```suggestion
        $componentExtensions = [
            'inviteComponent' => null,
            'resendInviteComponent' => null,
        ];
```

Or if you prefer we can already set the default here instead of doing the null checks below.

Reply:
This is awkward, because PHPStan isn't picking up the pass by reference into the events system, so thinks that the array can't change.

I've solved this by defining the defaults up front and setting an array shape on the original variable and the event listener.  There's nothing stopping someone from setting up a new event listener without the array shape - I imagine we can improve the event system to support typing, but that's out of the scope of this change.

Reply:
(A proper class would be nicer than an array shape, but that's not a pattern we currently use)